packages/truffle-core/lib/testing/deployed.js

```diff
@@ -2,7 +2,10 @@
 var Web3 = require("web3");
 
 var Deployed = {
-  makeSolidityDeployedAddressesLibrary: function(mapping) {
+  makeSolidityDeployedAddressesLibrary: function(
+    mapping,
+    removeAddressPayable
```

Comment:
Could we make this new second argument an "`options`" and use the whole solc compiler config for input?

Something like:
```javascript
makeSolidityDeployedAddressesLibrary: function (
  mapping,
  { solc: { version } }
) { /* ... */ }
```?

This way it can just be passed wholesale, and `testsource.js` wouldn't have to know what specific information it is that `deployed.js` needs.

If future versions of Solidity affect this file, the change will be self-contained here. And we don't really have a need to generalize what's different in different versions, so there's no need for it to be outside DeployedAddresses for the time being.

Reply:
Sounds good! 😎